codegen/type_converter.go

```diff
@@ -142,13 +144,14 @@ func (c *TypeConverter) genConverterForStruct(
 	fromPrefix string,
 	indent string,
 	fieldMap map[string]FieldMapperEntry,
+	prevKeyPrefixes []string,
 ) error {
-	toIdentifier := "out." + keyPrefix
-
-	typeName, err := c.getIdentifierName(toFieldType)
-	if err != nil {
-		return err
+	toIdentifier := "out"
+	if keyPrefix != "" {
+		toIdentifier += "." + keyPrefix
 	}
+
+	typeName, _ := c.getIdentifierName(toFieldType)
```

Comment:
probably shouldn't skip error check here. :)

Reply:
Right :) Will try to make the error testable / try to get value from a previous call

Reply:
It seems there's no way to get an error here, since c.getIdentifierName (or equivalent) is now called upstream, and it would error before it reaches here

Reply:
hmm seems the original code covered this case. Why difference?

Reply:
I added a ```typeName, err := c.getIdentifierName(toFieldType)``` in genStructConverter that wasn't there before, to check if that struct converter has already been generated / generate one if necessary

Reply:
ok. It's generally not the best practice to call the same function twice and skip error. Can we optimize that?

Reply:
I agree. I'd save the value and pass it down, but ```genConverterForStruct``` is also called from ```genConverterForMap``` and ```genConverterForList```, which already seem to have their own separate type validity checks, so I'm not able to produce an error from ```c.getIdentifierName(toFieldType)``` coming from the either of them (it errors beforehand)

Reply:
sure let's leave as it is.

Reply:
We should drop a comment here for why the error check is skipped.